src/pages/[guild]/events.tsx

```diff
@@ -133,39 +135,40 @@ const GuildEvents = (): JSX.Element => {
           <TabButton href={`/${urlName}/activity`}>Activity log</TabButton>
         )}
       </Tabs>
-      {!data && !error && isLoading && (
-        <FallbackFrame isLoading text="Searching for events..." />
-      )}
-      {!data && !error && !isLoading && (
-        <FallbackFrame
-          icon={NoteBlank}
-          title="No events yet"
-          text="Your guild has no upcoming event currently"
-        />
-      )}
-
-      {!isLoading && !!error && (
-        <FallbackFrame
-          icon={WarningOctagon}
-          text="Something went wrong, couldn't load events."
-        />
-      )}
-
-      {!isLoading && !error && data?.length === 0 && (
+      {isLoading ||
+        (data === undefined && (
+          <FallbackFrame isLoading text="Searching for events..." />
+        ))}
+      {!isLoading && data?.length === 0 && (
         <FallbackFrame
           icon={NoteBlank}
           title="No events yet"
           text="Your guild has no upcoming event currently or you have no access to Discord"
         />
       )}
-
-      {!isLoading && !error && data?.length > 0 && (
+      {!isLoading && data?.length > 0 && (
         <VStack gap={4}>
           {data.sort(sortEventByStartDate).map((event) => (
-            <DiscordEventCard key={event.id} event={event} guildId={guildId} />
+            <DiscordEventCard key={event.title} event={event} guildId={guildId} />
           ))}
         </VStack>
       )}
+      {isAdmin && !isLoading && error.length ? (
```

Comment:
nit: we could construct 1 error message inside `useGuildEvents` and just display that here, so we wouldn't need to do all this logic
(if this is blocking, let's refactor this later in a separate pr)